pingcap/tidb-operator
# *: implements RFC: Add 2021-11-24-graceful-reschedule-tikv-pod.md

### What problem does this PR solve?

implements #4277 
Fix #4215 
### What is changed and how does it work?

see docs in #4277 
### Code changes


### Tests


### test1
1. deploy a cluster (examples/basic/, change replicas of tikv to 3)
2. kubectl annotate pods basic-tikv-0 tidb.pingcap.com/restart=""
3. observe
    1. add evict-leader-scheduler
    2. tikv pod deleted
    3. remove evict-leader-scheduler
<img width="1522" alt="Screen Shot 2021-11-30 at 11 42 43 AM" src="https://user-images.githubusercontent.com/1681864/144002119-996f1405-80f2-4e41-b109-9cfcbd371269.png">

### test2
1. add `tidb.pingcap.com/restart` annotation, check evict-leader-scheduler added
2. remove the annotation and check evict-leader-scheduler removed

-- If this PR contains no code changes, check this box -->

### Side effects


### Related changes


### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Support evicting region leaders for TiKV and restarting TiKV gracefully through annotations
```
